docs/source/en/model_doc/propainter.md

```diff
@@ -0,0 +1,136 @@
+    '''
+    frames = []
+    container.seek(0)
+    start_index = indices[0]
+    end_index = indices[-1]
+    for i, frame in enumerate(container.decode(video=0)):
+        if i > end_index:
+            break
+        if i >= start_index and i in indices:
+            frames.append(frame)
+    return np.stack([x.to_ndarray(format="rgb24") for x in frames])
+
+
+def sample_frame_indices(clip_len, frame_sample_rate, seg_len):
+    '''
+    Sample a given number of frame indices from the video.
+    Args:
+        clip_len (`int`): Total number of frames to sample.
+        frame_sample_rate (`int`): Sample every n-th frame.
+        seg_len (`int`): Maximum allowed index of sample's last frame.
+    Returns:
+        indices (`List[int]`): List of sampled frame indices
+    '''
+    converted_len = int(clip_len * frame_sample_rate)
+    end_idx = np.random.randint(converted_len, seg_len)
+    start_idx = end_idx - converted_len
+    indices = np.linspace(start_idx, end_idx, num=clip_len)
+    indices = np.clip(indices, start_idx, end_idx - 1).astype(np.int64)
+    return indices
+
+##########################IF YOU HAVE THE MASK AND FRAMES IN THE FORM OF MP4(comment the below part if you want to use this)###########################
+# video clip consists of 80 frames(both masks and original video) (3 seconds at 24 FPS)
+video_file_path = hf_hub_download(
+    repo_id="ruffy369/propainter-object-removal", filename="object_removal_bmx/bmx.mp4", repo_type="dataset"
+)
+masks_file_path = hf_hub_download(
+    repo_id="ruffy369/propainter-object-removal", filename="object_removal_bmx/bmx_masks.mp4", repo_type="dataset"
+)
+container_video = av.open(video_file_path)
+container_masks = av.open(masks_file_path)
+
+
+# sample 32 frames
+indices = sample_frame_indices(clip_len=32, frame_sample_rate=1, seg_len=container_video.streams.video[0].frames)
+video = read_video_pyav(container=container_video, indices=indices)
+
+masks = read_video_pyav(container=container_masks, indices=indices)
+video = list(video)
+masks = list(masks)
+
+
+##########################IF YOU HAVE FOLDER WITH JPG IMAGES OF FRAMES AND MASKS(comment the above part if you want to use this########################
+from datasets import load_dataset
+import numpy as np
+ds = load_dataset("ruffy369/propainter-object-removal")
+ds_images = ds['train']["image"]
+num_frames = 80
+video = [np.array(ds_images[i]) for i in range(num_frames)]
+#stack to convert H,W mask frame to compatible H,W,C frame
+masks = [np.stack([np.array(ds_images[i])] * 3, axis=-1) for i in range(num_frames, 2*num_frames)]
+
+####################################################START OF THE IMAGE PROCESSOR AND MODEL FORWARD PASS################################################
+from transformers import ProPainterImageProcessor
+image_processor = ProPainterImageProcessor()
+inputs = image_processor(images = video, masks = masks)
```

Comment:
@amyeroberts Are you talking about creating a separate `video_processing_` file instead of image_processing and create `VideoProcessor` there and change the arg name from images to videos there? Like how it's been done in this [PR](https://github.com/huggingface/transformers/pull/32673) ?!

Reply:
Update: I done the same in the latest commits

Reply:
> @amyeroberts Are you talking about creating a separate `video_processing_` file instead of image_processing and create `VideoProcessor` there and change the arg name from images to videos there? Like how it's been done in this [PR](https://github.com/huggingface/transformers/pull/32673) ?!

This was the question I wanted to ask. Although I have pushed the commits in the same way that you mentioned